tests/run-with-compiler/i14541.scala

```diff
@@ -0,0 +1,14 @@
+
+// test argument processing and "execution mode" in runner
+object Test:
+  import dotty.tools.runner.RichClassLoader.*
+  val classpath = dotty.tools.dotc.util.ClasspathFromClassloader(getClass.getClassLoader)
+  def main(args: Array[String]): Unit =
+    getClass.getClassLoader.run("echo", List("hello", "raw", "world"))
+    getClass.getClassLoader.run("dotty.tools.MainGenericRunner", List("echo", "hello", "rawr", "world"))
```

Comment:
This call works after using "parent class loader" in Runner. Folks with an opinion about how the command line runner should behave might have an opinion.